drivers/retained_mem/Kconfig.nrf

```diff
@@ -8,3 +8,11 @@ config RETAINED_MEM_NRF_GPREGRET
 	help
 	  Enable driver for Nordic nRF GPREGRET-based retained memory
 	  register support.
+
+config RETAINED_MEM_NRF_RAM_CTRL
+	bool "nRF RAM retention driver"
+	default y
+	depends on DT_HAS_ZEPHYR_RETAINED_RAM_ENABLED && RETAINED_MEM_ZEPHYR_RAM
+	depends on SOC_SERIES_NRF52X || SOC_SERIES_NRF53X || SOC_SERIES_NRF54LX || SOC_SERIES_NRF91X
```

Comment:
`depends on SOC_SERIES_NRF52X || SOC_SERIES_NRF53X || SOC_SERIES_NRF54LX || SOC_SERIES_NRF91X`

What? Why is this needed on anything that isn't nrf54l?

Reply:
Purpose of this PR was to mainly support nrf54l, however the updated `retained_mem` driver supports nrf52/nrf53/nrf91 as well, hence this list of ORed SoC series

Reply:
Support how? The RAM is retained by default which is what I assume `nrfx_ram_ctrl_retention_enable_set` sets, so why is that needed when it's already enabled?

Reply:
All RAM is enabled by default, I don't mean retention I just mean RAM itself, which means you can reboot an application and keep the contents, this definitely works because we have 2 tests that check it on real hardware in https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/boot/mcuboot_data_sharing and https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/boot/mcuboot_recovery_retention

So this allows retaining RAM contents if someone turns the RAM section off... but why? If someone turns RAM sections off, they should know what they are doing, and having a part that is retained that you can't do anything with makes no sense to me. Not to mention that as I found out and posted on NCSDK-22549 after a gruelling battle with an ncs sample application and debugger, turning off RAM sections is a great way to lock the whole CPU up - or brick the module.

Reply:
This is about preserving partial RAM content when using https://docs.zephyrproject.org/apidoc/latest/group__sys__poweroff.html#gaf4552a8ea40b4adb94f85b0a1ff06e39

This function results in SystemOff state in nRF52/53/91 in which the RAM is powered down and its content is unretained by default.

Reply:
Interesting, then this should depends on `POWEROFF`

Reply:
added dependency on `POWEROFF`

Reply:
you have added the wrong dependency, it needs to be `POWEROFF` not `HAS_POWEROFF`